src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -185,24 +185,17 @@ private function buildChildDefinitions(ContainerBuilder $container, string $seri
 
             $definition = $container->registerChild($childId, (string) $id);
 
-            if (null !== $nameConverterIndex = $this->findNameConverterIndex($container, (string) $id)) {
-                $definition->replaceArgument($nameConverterIndex, new Reference($config['name_converter']));
+            foreach ($container->getDefinition($id)->getArguments() as $index => $argument) {
+                if ($argument instanceof Reference && self::NAME_CONVERTER_METADATA_AWARE_ID === (string) $argument) {
+                    $definition->replaceArgument($index, new Reference($config['name_converter']));
+                } elseif ($argument instanceof Reference && str_starts_with((string) $argument, 'serializer.normalizer.')) {
```

Comment:
@ruudk This won't work since the service is not tagged with `serializer.normalizer`:
https://github.com/symfony/symfony/blob/ac26e1b576034328a7191f79fdc3a0368a00c178/src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.php#L136-L146

Without the tag, the child service will never be created. As far as I can tell, the service was intentionally not tagged, but I could be wrong, see #37847.

Reply:
Weird because this fix does solve the problem.

Reply:
Are you sure? I've tested the fix on a clean install and got an error:

```yaml
framework:
    serializer:
        named_serializers:
            foo:
                name_converter: serializer.name_converter.camel_case_to_snake_case
```

```
The service "serializer.normalizer.mime_message.foo" has a dependency on a non-existent service "serializer.normalizer.property.foo".
```

Reply:
Oops, you're right. For some reason, the project where we experience this problem, had the following compiler pass:

```php
 $container->addCompilerPass(
            new class() implements CompilerPassInterface {
                public function process(ContainerBuilder $container) : void
                {
                    $container->getDefinition('serializer.normalizer.property')
                        ->addTag('serializer.normalizer', [
                            'priority' => -990,
                        ]);
                }
            },
            priority: 10,
        );
```

I cannot recall why we added this, but probably to make sure serialization worked the same way as JMS Serializer.

Reply:
I know why we added this, because we want to use the PropertyNormalizer to make sure no getters are called while serializing. We only want properties to be serialized. 

Is there a better way to enable this? Or is the compiler pass above the recommended way to use this normalizer?

Maybe we can check in this PR if the `serializer.normalizer.*.$serializer_name` exists, and then replace it?

Reply:
@HypeMC Any idea how to proceed?

Reply:
I tried fixing this a while back (#60091) but ran into another issue. I haven't had time since, but I'm planning to look into it, hopefully this week.

Reply:
That would be great! Thanks 🙏🏻